.github/workflows/docker-build.yaml

```diff
@@ -0,0 +1,56 @@
+---
+# build docker image, push to registry
+
+name: Build and Push Docker Image
```

Comment:
Remind me of the advantages in building here and pushing the image instead of on dockerhub?

Reply:
I did a research cycle on registry options we could push to, including initially Dockerhub and AWS ECR. Dockerhub was the choice I'd reach for previously, but I still don't see a great way to push without an explicit user account, and the free tier for orgs gives us only 3 seats, all filled. A service/bot account is what I would use for this automation, with permissions scoped appropriately.

Github Container Registry in its current form [appears new as of earlier this year](https://github.blog/2021-06-21-github-packages-container-registry-generally-available/). It wasn't on my radar before, but it solves the seat issue and avoids Dockerhub's other free tier restrictions (e.g. number of repos). For open source projects it (currently) gives us everything we need for free, so seems just as good if not better a distribution hub for container images as Dockerhub would be.